openssl/src/asn1.rs

```diff
@@ -165,6 +171,28 @@ impl Asn1Time {
     }
 }
 
+impl Asn1TimeRef {
+    /// Returns the duration between `older` and `self`.
+    ///
+    /// If `self` represents an earlier time than `older`, or `self` or `older` are invalid, or
+    /// the duration between `older` and `self` does not fit in a `Duration`, this will return
+    /// `None`.
+    pub fn duration_since(&self, older: &Self) -> Option<Duration> {
+        unsafe {
+            let mut days = 0;
+            let mut secs = 0;
+            cvt(ffi::ASN1_TIME_diff(&mut days, &mut secs, older.as_ptr(), self.as_ptr())).ok()?;
+
+            if days < 0 || secs < 0 {
+                None
+            } else {
+                secs = secs.checked_add(days.checked_mul(24 * 60 * 60)?)?;
```

Comment:
It seems wrong that this assumes that all days are the same length.

Reply:
Yes, however `days_from_now` does the same, and OpenSSL also seems to use [this `SECS_PER_DAY`](https://github.com/openssl/openssl/blob/0e9725bcb90770d967351b977407b174bbd91869/crypto/o_time.c#L65) constant for the conversion

Reply:
Generally I like to provide a fairly direct binding to the underlying APIs - in this case that'd be returning a struct of the days and seconds. The days_from_now constructor is a bit of a holdover from an earlier time that I should probably clean up.